papiea-engine/src/databases/provider_db_mongo.ts

```
@@ -65,7 +65,18 @@ export class Provider_DB_Mongo implements Provider_DB {
         if (intentful_kinds.length === 0) {
             return
         }
-        await this.subCollection.insertMany(intentful_kinds)
+        try {
+            await this.subCollection.insertMany(intentful_kinds)
+        } catch (err) {
+            // Check a DuplicateMongo Error
+            // If the error is a DuplicateMongo Error
+            // then we are reregestering provider and its fine
+            if (err.code === 11000) {
+                return
```

Is this correct? Is the entity getting updated and throws the exception after a successful update? This looks only like ignoring the exception.

---

```
            await this.subCollection.createIndex({
                "provider_prefix": 1,
                "provider_version": 1,
                "kind_name": 1
            }, { unique: true })
```
This the piece of code that throws since we are making a unique collection of intenful kinds and `save_provider` tries to do that once again. Considering that kinds didn't change it is safe that we keep the old index, if they had changed then it is not going to raise an Exception since a new set of unique pair is going to be generated

---

hmm.. why are we recreating the index? Isn't mongo updating the index when content changes?

---

I might be a bit wrong on this one but Mongo doesn't really modify the index at all, you need to reindex for that (drop -> recreate)